src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java

```diff
@@ -332,13 +318,12 @@ <U> List<T> select(
             int first,
             int numResults) {
         final RelationalDaoPriv dao = daos.get(context.getShardId());
-        SelectParamPriv selectParam = SelectParamPriv.builder()
-                .criteria(criteria)
-                .start(first)
-                .numRows(numResults)
-                .build();
-        return transactionExecutor.execute(context.getSessionFactory(), true, dao::select, selectParam, t -> t, false,
-                                           "select", context.getShardId());
+        val opContext = Select.<T, List<T>>builder()
+            .getter(dao::select)
+            .criteria(criteria)
+            .start(first)
+            .numRows(numResults).build();
+        return transactionExecutor.execute(context.getSessionFactory(), true, "select", opContext, context.getShardId());
```

Comment:
completeTransaction should be false here

Reply:
Done.